### PR TITLE
[CategoryAxis] Right oriented category axis respects tick marks now

### LINKED ISSUE #1646: "right" oriented Category Axis doesn't respect tick-marks
screen shot 2015-02-17 at 3 43 13 pm

http://jsfiddle.net/j1af17uu/3/
The label text is on top of the tick marks (big magenta blobs)

..."left" oriented for comparison http://jsfiddle.net/j1af17uu/4/

### PR DESCRIPTION
**Pull request by rcchen**
Fixes #1646

### CODE REVIEW

File: src/components/axes/categoryAxis.ts [hunk 1]
```diff
@@ -178,7 +178,7 @@ export module Axis {
 
       var xTranslate = this.orient() === "left" ? 0 : this._maxLabelTickLength() + this.tickLabelPadding();
       var yTranslate = this.orient() === "bottom" ? this._maxLabelTickLength() + this.tickLabelPadding() : 0;
-      _Util.DOM.translate(this._tickLabelContainer, 0, yTranslate);
```
Comment (jtlan) [inline src/components/axes/categoryAxis.ts]: wat

Comment (bluong) [inline src/components/axes/categoryAxis.ts]: What happens if the orient is bottom?  wouldn't the tickLabels be shifted to the right for no good reason?

Comment (rcchen) [inline src/components/axes/categoryAxis.ts]: Good catch @bluong

File: src/components/axes/categoryAxis.ts [hunk 2]
```diff
@@ -176,12 +176,15 @@ export module Axis {
...
+      var xTranslate = this._isOriented(["top", "bottom", "left"]) ? 0 : this._maxLabelTickLength() + this.tickLabelPadding();
+      var yTranslate = this._isOriented(["bottom"]) ? this._maxLabelTickLength() + this.tickLabelPadding() : 0;
+      _Util.DOM.translate(this._tickLabelContainer, xTranslate, yTranslate);
       return this;
     }
 
+    private _isOriented(orientations: string[]) {
+      return orientations.some((orientation: string) => this.orient() === orientation);
+    }
```
Comment (bluong) [inline src/components/axes/categoryAxis.ts]: Not seeing the difference of this and indexOf > -1

File: src/components/axes/categoryAxis.ts [hunk 3]
```diff
@@ -176,12 +176,15 @@ export module Axis {
       this._drawTicks(this.width(), this.height(), ordScale, tickLabels);
       var translate = this._isHorizontal() ? [ordScale.rangeBand() / 2, 0] : [0, ordScale.rangeBand() / 2];
 
-      var xTranslate = this.orient() === "left" ? 0 : this._maxLabelTickLength() + this.tickLabelPadding();
-      var yTranslate = this.orient() === "bottom" ? this._maxLabelTickLength() + this.tickLabelPadding() : 0;
-      _Util.DOM.translate(this._tickLabelContainer, 0, yTranslate);
+      var xTranslate = this._isOriented(["top", "bottom", "left"]) ? 0 : this._maxLabelTickLength() + this.tickLabelPadding();
```
Comment (bluong) [inline src/components/axes/categoryAxis.ts]: You could just do a strict check for right and 0 otherwise

**Pull request merged**